### ISSUE TITLE
Functional calculations and stdev setup issues

### ISSUE THREAD

**Issue by zhangweiteng** [SMTG-Bham/ShakeNBreak#84]
In the process of calculating oxides with SnB distortion, I encountered two questions that needed to be answered. If I can, I would like to express my sincere gratitude. 
. It's the problem of stdev settings, because the structure I chose is oxide ionic type, and I can't follow the default 10%, is my stdev setting for all defect states or the defects in the Tricky Relaxations part reset stdev 7.5% or 5% to do it? There are also different stdevs that can be compared in the same transition level diagram.

**Comment by ireaml** [comment]
Hi @zhangweiteng , thanks for the interest in ShakeNBreak.
I'm not sure if I fully understand your issue. The recommended approach (see this [wiki section](https://shakenbreak.readthedocs.io/en/latest/Tips.html#hard-ionic-magnetic-materials)) is to first use the default rattle `stdev` (`SnB` uses the same `stdev` value for all defects). If you suspect potential issues, you could first test it for one defect. If the relaxations are taking too long and not converging, `SnB` will warn you. If so, you should reduce the `stdev` to a smaller value like 5%. 

Have you already tried using the default `stdev` and encountered issues? What material are you studying?

**Comment by zhangweiteng** [comment]
The result I got with the default setting is not quite right, and the current setting of 7.5% of the same INCAR does not converge, and I am still looking for the reasonThe image below is a warp problem by default.![Image](https://github.com/user-attachments/assets/113955d1-b2ba-429d-b63e-8e6d01f99ed8)          
 I can judge whether the warp failed based on such content.

**Comment by zhangweiteng** [comment]
Hi！@ireaml I'm spending too much time calculating the twisted structure with HSE06 and may get untwisted results. After reading this article https://doi.org/10.1038/s41524-024-01303-9 I was thinking, can I use PBE or PBEsol's low-cost algorithm to complete this process of screening the ground state, and after determining the ground state structure, I can calculate HSE06 again in two ways? One is to directly take the structure obtained by PBE and perform vasp_std subsequent calculations, and the other is to re-twist the determined ground state object and then perform the vasp_std and other calculations.

**Comment by kavanase** [comment]
Hi @zhangweiteng,
Your question is not totally clear, but for the image you copied above:
![Image](https://github.com/user-attachments/assets/31da5d73-6751-4b4a-83ea-d1374c4a379d)
This does not indicate any problem with rattling for those cases, but just that no _lower energy structure_ was found from rattling & distortions, as compared to an unperturbed relaxation (with no initial distortions). If there is problems with rattling (where you may need to adjust `stdev`), you will receive a warning (as @ireaml said) like this:
https://github.com/SMTG-Bham/ShakeNBreak/blob/bed6e0f08a218e16735db050a281c931c3f95a0e/shakenbreak/io.py#L294-L304
(see discussion here: https://shakenbreak.readthedocs.io/en/latest/Tips.html#hard-ionic-magnetic-materials).

For your second question, we typically recommend not using GGA DFT for structure searching as it can be inaccurate due to issues with charge localisation and band gap underestimation, which severely affect defect predictions. In the paper you link, M3GNet was trained on hybrid DFT data. Usually, there are other choices that can be made to make calculations run faster, such as using an appropriate supercell (sufficiently large, but not excessively large -- the [`doped`](https://doped.readthedocs.io/en/latest/) supercell generation algorithm can be good for this), appropriate `ENCUT` (typically this can be reduced by ~20-30% from the fully-converged value while still giving sufficiently accurate forces/energies for SnB, appropriate `POTCAR`s, good HPC optimisation (`NCORE` etc). You could also reduce the density of the SnB distortion mesh (using `distortion_increment` -- see API docs). If these do not make the calculations fast enough, then SnB with GGA DFT is better than not doing it at all.

**Comment by zhangweiteng** [comment]
Hi! @kavanase 
Sorry for the trouble. My neutral valence distorted energy distribution in the oxygen vacancy shows that the undistorted structure energy appears to be lower than the fully distorted structure. So I tried lowering the stdev setting and searching to find the effect of the lowest structure.
![Image](https://github.com/user-attachments/assets/6de08597-b320-435e-acf1-57076ad5dc32)
I found that the calculation of other defective structures will have the same warning message if the calculation is not completed, so I will complete the further calculation of the twisted structure.

For the second question, I also performed PEBsol and HSE06 functional tests on my structure, and found that the results were indeed too different, and the database of HSE06 functionals was used in the link paper to fine-tune and other forms to achieve approximately equal HSE06 effects. Thank you for your valuable reply, and thank you for being able to accommodate my rough insights.

**Comment by kavanase** [comment]
Hi @zhangweiteng ,
The energy differences between your distorted calculations and unperturbed are about 0.7 meV, which is negligible (i.e. they are basically the same). This is why you don't receive any warnings. There is some small residual noise in the energies of the distorted calculations due to the use of coarse relaxation parameters to make the calculations run fast but with sufficient qualitative accuracy, which is why they are very slightly higher in energy here. We usually don't consider energy differences from these calculations to be significant until at least ~20 meV, if not higher. So there are no problems with these calculations and it looks like the automatic `stdev` from ShakeNBreak is performing perfectly fine here.

**Comment by zhangweiteng** [comment]
Hi@kavanase ，
Thank you very much for your valuable reply! I also completed a clear understanding of this issue. Thank you again for your analysis and guidance.
I wish ShakeNBreak will get better and better!
